compiler/rustc_pattern_analysis/src/lib.rs

```diff
@@ -2,6 +2,7 @@
 
 #![allow(rustc::untranslatable_diagnostic)]
 #![allow(rustc::diagnostic_outside_of_impl)]
+#![feature(freeze)]
```

Comment:
If this actually gets merged, please set this only when `feature = "rustc"`. This crate needs to build on stable for rust-analyzer.

Reply:
is it needed just for this? https://github.com/rust-lang/rust/blob/94b72d6bebbddb5d17860914c4544f37365e988c/compiler/rustc_pattern_analysis/src/pat.rs?plain=1#L250

Reply:
This is an experiment, it won't get merged. Also the crate can't build on stable any more with this rustc change.

Reply:
> is it needed just for this? 

Yes,